doc/Language-Reference.md

```diff
@@ -117,15 +124,23 @@ A message type specifies the message format of a protocol. Each component corres
 ```Ada RFLX declaration
 type Frame is
    message
-      Destination : U48;
-      Source : U48;
-      EtherType : U16
+      Destination : Address;
+      Source : Address;
+      Type_Length_TPID : Type_Length
+         then TPID
+            with First => Type_Length_TPID'First
+            if Type_Length_TPID = 16#8100#
          then Payload
-            with Size => EtherType * 8
-            if EtherType <= 1500
+            with Size => Type_Length_TPID * 8
+            if Type_Length_TPID <= 1500
+         then Ether_Type
+            with First => Type_Length_TPID'First
+            if Type_Length_TPID >= 1536 and Type_Length_TPID /= 16#8100#;
+      TPID : TPID;
+      TCI : TCI;
+      Ether_Type : Ether_Type
          then Payload
-            with Size => Message'Last - EtherType'Last
-            if EtherType >= 1536;
+            with Size => Message'Last - Ether_Type'Last;
       Payload : Opaque
```

Comment:
I suppose you are asking for a general rule here. It seems not to be defined in our Ada-Style guide. I'm not sure if aligning colons makes sense in all cases. It could be probably useful for smaller message specifications, for bigger ones (like [HTTP/2 frames](https://github.com/Componolit/RecordFlux-specifications/blob/main/http_2.rflx)) it does not seem reasonable to me. @senier: What do you think?

For comparison:

```Ada
   type Frame is
      message
         Destination : Address;
         Source : Address;
         Type_Length_TPID : Type_Length
            then TPID
               with First => Type_Length_TPID'First
               if Type_Length_TPID = 16#8100#
            then Payload
               with Size => Type_Length_TPID * 8
               if Type_Length_TPID <= 1500
            then Ether_Type
               with First => Type_Length_TPID'First
               if Type_Length_TPID >= 1536 and Type_Length_TPID /= 16#8100#;
         TPID : TPID;
         TCI : TCI;
         Ether_Type : Ether_Type
            then Payload
               with Size => Message'Last - Ether_Type'Last;
         Payload : Opaque
            then null
               if Payload'Size / 8 >= 46 and Payload'Size / 8 <= 1500;
      end message;
```

```Ada
   type Frame is
      message
         Destination : Address;
         Source      : Address;
         Type_Length_TPID : Type_Length
            then TPID
               with First => Type_Length_TPID'First
               if Type_Length_TPID = 16#8100#
            then Payload
               with Size => Type_Length_TPID * 8
               if Type_Length_TPID <= 1500
            then Ether_Type
               with First => Type_Length_TPID'First
               if Type_Length_TPID >= 1536 and Type_Length_TPID /= 16#8100#;
         TPID        : TPID;
         TCI         : TCI;
         Ether_Type  : Ether_Type
            then Payload
               with Size => Message'Last - Ether_Type'Last;
         Payload     : Opaque
            then null
               if Payload'Size / 8 >= 46 and Payload'Size / 8 <= 1500;
      end message;
```

Reply:
> It seems not to be defined in our Ada-Style guide.

It is defined in the original [Ada Style Guide](https://upload.wikimedia.org/wikipedia/commons/7/7c/Ada_Style_Guide.pdf) which out style guide builds upon (although I noticed that we didn't write this down anywhere).

I would also be fine with aligning only blocks. I see that it doesn't make sense when there are large conditions in between.

Reply:
Generally, I like the alignment of colons mandated by the Ada Style Guide. However, our message specifications are different from simple Ada records (variant records are comparable, but I couldn't find a layout example in the style guide). I'd align consecutive fields as suggested by @treiher (we need to decide whether `Source` and `Type_Length_TPID` are consecutive - I'd say they are).

Reply:
I'm not convinced by this block-wise alignment. IMHO it gives the impression that consecutive fields without explicit link are more related to each other than to other fields, which is not the case.

Reply:
That's a good point. Then we should not align - if you have a look at `ICMP::Message` then alignment doesn't seem to make sense.

Reply:
@jklmnn Are you OK with not aligning? If so, please resolve this conversion.

Reply:
It's okay for me.